backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/LibvirtVmXmlBuilder.java

```diff
@@ -2437,21 +2437,22 @@ private void writeDiskSource(VmDevice device, Disk disk, String dev, DiskVmEleme
         case MANAGED_BLOCK_STORAGE:
             ManagedBlockStorageDisk managedBlockStorageDisk = (ManagedBlockStorageDisk) disk;
             Map<String, String> metadata = new HashMap<>();
-            String path = (String) managedBlockStorageDisk.getDevice().get(DeviceInfoReturn.PATH);
+            String path;
+            if (Version.v4_7.lessOrEquals(vm.getCompatibilityVersion())) {
+                path = (String) managedBlockStorageDisk.getDevice().get(DeviceInfoReturn.MANAGED_PATH);
+            } else {
+                path = (String) managedBlockStorageDisk.getDevice().get(DeviceInfoReturn.PATH);
+            }
```

Comment:
How does this work for running vm when we upgrade the cluster version while the vm is running?
 
The vm will start using "path". After the upgrade, if we migrate the vm, we will use "managed_path", and the migration will likely fail.

Can we always use "managed_path"?

Reply:
what about older cluster levels, if we do live migration in a 4.6 cluster for example?

Reply:
Migration to/from 4.6 cluster will not work with managed_path. But since this is tech preview we don't need to care about it.

If we want to avoid this issue we can use managed_path only for lightos, and continue to use path for rbd and mpath. But this means we carry to ways to work with mbs storage, which is bad. I this to break this now and not later.

Reply:
Now it is clear, we use manage_path for cluster level 4.7, and path for older cluster levels since vdsm in older versions did not support managed_path.